service/storage_service.cc

```diff
@@ -5384,7 +5385,14 @@ future<> storage_service::process_tablet_split_candidate(table_id table) noexcep
         } catch (raft::request_aborted& ex) {
             slogger.warn("Failed to complete splitting of table {} due to {}", table, ex);
             break;
+        } catch (seastar::rpc::remote_verb_error& ex) {
```

Comment:
The proper fix is to abort/stop the fibers which rely on group0 membership as part of the decommission process.

Reply:
We're stopping the fibers when stopping group0 (see storage_service::wait_for_group0_stop()), when shutting down the service.

In order for it to work properly, we need to deregister such fibers before we actually proceed to remove node membership, but what if this latter fails? Can it happen we revert decommission at such late stage and the node will remain with dead fibers?

Reply:
Maybe it's enough to do _group0_as.request_abort() right after group0 membership removal succeeds, since fibers are prepared to handle abort_requested_exception. So termination of fibers can be done as today when shutting down group0 service. They're conceptually coupled. What do you think?

Reply:
I'd have to go through all that shudown path in storage service to make sure I am not missing anything. For example, is there any process post membership removal that relies on group0_as not being aborted?

Reply:
> We're stopping the fibers when stopping group0 (see storage_service::wait_for_group0_stop()), when shutting down the service.
> 
> In order for it to work properly, we need to deregister such fibers before we actually proceed to remove node membership, but what if this latter fails? Can it happen we revert decommission at such late stage and the node will remain with dead fibers?

It's part of graceful shutdown but not mandatory, so if this RPC fails we can still commit decommission.

> Maybe it's enough to do _group0_as.request_abort() right after group0 membership removal succeeds

Where do we remove node from group0 membership on decommission?